src/event.rs

```diff
@@ -589,38 +599,77 @@ pub enum DeviceEvent {
         state: ElementState,
     },
 
-    Key(KeyboardInput),
+    Key(RawKeyEvent),
 
     Text {
         codepoint: char,
     },
 }
 
-/// Describes a keyboard input event.
-#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+/// Describes a keyboard input as a raw device event.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct KeyboardInput {
-    /// Identifies the physical key pressed
-    ///
-    /// This should not change if the user adjusts the host's keyboard map. Use when the physical location of the
-    /// key is more important than the key's host GUI semantics, such as for movement controls in a first-person
-    /// game.
-    pub scancode: ScanCode,
-
+pub struct RawKeyEvent {
+    pub physical_key: keyboard::KeyCode,
     pub state: ElementState,
+    pub repeat: bool,
```

Comment:
What's the motivation for adding a `repeat` field here? It seems to be purely emulated right now.

Reply:
Yeah it is emulated, but as far as I know, this is the best anyone can do with raw device events, and the reasoning was that most applications using this event will probably want to be able differentiate between repeat and non-repeat. But I'm not knowledgeable in how this sort of event is usually used, so if this is not relevant here, then it can be removed.

Reply:
I'm leaning towards removing it. In particular for gamedev, key repeat is less important as you typically manually track only if a key is pressed or not

Reply:
I'd be ok with removing it, even if it's less convenient, but I'd like for the documentation of `RawKeyEvent` to mention the expected behaviour if the `repeat` field is removed.